lib/driver.js

```diff
@@ -1075,10 +1079,131 @@ export class XCUITestDriver extends BaseDriver {
   }
 
   /**
-   *
-   * @param {string} cmd
-   * @param {...any} args
-   * @returns {Promise<any>}
+   * Checks if an error is related to WDA connection issues
+   * @param {Error|null|undefined} err - The error to check
+   * @returns {boolean} - True if the error is a WDA connection error
+   */
+  isWdaConnectionError(err) {
+    if (!err) {
+      return false;
+    }
+    const errorMessage = err.message || '';
+    const CONNECTION_ERRORS = ['ECONNREFUSED', 'socket hang up'];
+
+    return CONNECTION_ERRORS.some((errStr) => errorMessage.includes(errStr));
+  }
+
+  /**
+   * Reconnects to WDA within a specified timeout
+   * @returns {Promise<object>} - The WDA status
+   * @throws {Error} - If WDA does not become reachable within the timeout
+   */
+  async reconnectWda() {
+    const timeout = this.opts.wdaConnectionTimeout || 15000;
+    const interval = this.opts.wdaStartupRetryInterval || 1000;
+    const startTime = Date.now();
+
+    this.log.info(`Attempting to reconnect to WDA with ${timeout}ms timeout`);
+
+    while (Date.now() - startTime < timeout) {
```

Comment:
use primitives from the `asyncbox` module from this purpose, e.g. waitForCondition